extensions/default/src/DicomWebDataSource/utils/clientManager.ts

```diff
@@ -0,0 +1,153 @@
+import { api } from 'dicomweb-client';
+import StaticWadoClient from './StaticWadoClient';
+import dcm4cheeReject from '../dcm4cheeReject';
+
+import { errorHandler, utils } from '@ohif/core';
+
+export default class clientManager {
+  clients;
+  userAuthenticationService;
+
+  // adds a dicomweb server configuration in the clients list
+  public addConfiguration(params, query, config) {
+    if (config.onConfiguration && typeof config.onConfiguration === 'function') {
+      config = config.onConfiguration(config, {
+        params,
+        query,
+      });
+    }
+
+    config.qidoConfig = {
+      url: config.qidoRoot,
+      staticWado: config.staticWado,
+      singlepart: config.singlepart,
+      headers: this.userAuthenticationService.getAuthorizationHeader(),
+      errorInterceptor: errorHandler.getHTTPErrorHandler(),
+    };
+
+    config.wadoConfig = {
+      url: config.wadoRoot,
+      staticWado: config.staticWado,
+      singlepart: config.singlepart,
+      headers: this.userAuthenticationService.getAuthorizationHeader(),
+      errorInterceptor: errorHandler.getHTTPErrorHandler(),
+    };
+
```

Comment:
We should allow the creation of multiple instances of a single data source at the OHIF level instead of adding this capability at the data source level. This way later we can gradually add support for other data sources to have multiple instances If the configuration is an array. @sedghi what do you think?

Reply:
I done that to explicitly define the servers a data source is supposed to work.

Reply:
Anyway, as there will be the case for defining the second server via link we can leave the definitions as is, as we can define a normal configuration and add the second via link